app/models/developers/external_profiles/linkedin.rb

```diff
@@ -0,0 +1,54 @@
+module Developers::ExternalProfiles
+  class Linkedin
+    def initialize
+      @api_key = api_key
+      @endpoint = endpoint
+    end
+
+    def get_profile(url)
+      params = {
+        "url" => url,
+        "fallback_to_cache" => "on-error",
+        "use_cache" => "if-present"
+      }
+
+      conn = Faraday.new(@endpoint) do |faraday|
+        faraday.headers["Authorization"] = "Bearer #{@api_key}"
+        faraday.adapter Faraday.default_adapter
+        faraday.response :json
+      end
+
+      begin
+        response = conn.get("", params)
+        if response.status.to_i == 200
+          parse_json_response(response.body)
+        else
+          {error: "API Error: #{response.status} - #{response.body}"}
```

Comment:
I'd like to return a concrete type here instead of a hash.

```ruby
module Developers::ExternalProfiles
  class Linkedin
    class Response
      attr_reader :error, :data

      def initialize(data: nil, error: nil)
        @data = data
        @error = error
      end

      def error?
        error.present?
      end
    end
  end
end
```

Then we can make this change, which I think cleans things up a bit:

```diff
diff --git a/app/models/developers/external_profiles/linkedin_profile_fetcher.rb b/app/models/developers/external_profiles/linkedin_profile_fetcher.rb
index 29682b62..9b527370 100644
--- a/app/models/developers/external_profiles/linkedin_profile_fetcher.rb
+++ b/app/models/developers/external_profiles/linkedin_profile_fetcher.rb
@@ -12,13 +12,13 @@ module Developers::ExternalProfiles
     end

     def external_profile(developer, response)
-      if response[:error]
-        {developer_id: developer.id, site: "linkedin", data: {}, error: response[:error]}
+      if response.error?
+        {developer_id: developer.id, site: "linkedin", data: {}, error: response.error}
       else
         external_profile = Developers::ExternalProfile.linkedin_developer(developer)
-        external_profile.data = response[:data] unless external_profile.blank?
+        external_profile.data = response.data unless external_profile.blank?
         if external_profile.blank? || external_profile.data_changed?
-          {developer_id: developer.id, site: "linkedin", data: response[:data], error: nil}
+          {developer_id: developer.id, site: "linkedin", data: response.data, error: nil}
         end
       end
     end
```